AWP-1634: Add Header component

global-usman opened the pull request:
[AWP-1634](https://team-1637335847501.atlassian.net/browse/AWP-1634)

- Created Header component with subcomponents: Heading, Button, and Breadcrumbs
- Implemented styled components for Header elements
- Added Storybook stories for Header component with various configurations
- Created unit tests for Header component to ensure proper rendering of subcomponents

[AWP-1634]: https://team-1637335847501.atlassian.net/browse/AWP-1634?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

global-usman committed:
AWP-1634: Add Header component with styled subcomponents and Storybook examples

- Created Header component with subcomponents: Heading, Button, and Breadcrumbs
- Implemented styled components for Header elements
- Added Storybook stories for Header component with various configurations
- Created unit tests for Header component to ensure proper rendering of subcomponents

darnold001 reviewed (approved):
LGTM

zamanaf commented on src/components/header/Header.tsx:
```diff
@@ -0,0 +1,73 @@
+  let heading: ReactNode | null = null;
+  let breadcrumbs: ReactNode | null = null;
+
+  React.Children.forEach(children, (child) => {
+    if (React.isValidElement(child)) {
+      if (child.type === Button) {
+        buttons.push(child);
+      } else if (child.type === HeadingComponent) {
+        heading = child;
+      } else if (child.type === Breadcrumbs) {
+        breadcrumbs = child;
+      }
+    }
+  });
```
I think we can get rid of this by leveraging the useSlot hook. in the slideBar PR I have.

zamanaf reviewed (approved):
LGTM

global-usman committed:
Refactor Header component to utilize the useSlots hook for better organization

- Utilize useSlots hook to manage child components dynamically.
- Update component structure to improve readability and maintainability.
- Adjust styling for button container alignment.
- Update storybook examples and documentation.
